Demo/RCHSearchViewController.swift

```diff
@@ -19,30 +19,56 @@ class RCHSearchViewController: UIViewController, UISearchBarDelegate, UICollecti
     @IBOutlet weak var searchProductsImageView: UIImageView!
     @IBOutlet weak var autocompleteTableView: UITableView!
     
-    var productArray: [RCHRecommendedProduct] = []
-    var autocompleteArray: [String] = ["Boots", "Bomber", "Boucle", "Boyfriend"]
-    var searchTerm = ""
+    var products: [RCHSearchProduct]  = [] {
+        didSet {
+            searchResultsCollectionView.reloadData()
+            if products.isEmpty {
+                searchResultsCollectionView.isHidden = true
+                searchProductsView.isHidden = false
+                searchProductsLabel.text = "No Results"
+                searchProductsImageView.image = UIImage(named: "icn-tabbar-shop.pdf")
+            } else {
+                searchResultsCollectionView.isHidden = false
+                searchProductsView.isHidden = true
+            }
+        }
+    }
+    var autocompleteSuggestions: [String] = [] {
+        didSet {
+            autocompleteTableView.reloadData()
+            if autocompleteSuggestions.isEmpty {
+                autocompleteTableView.isHidden = true
+                searchProductsLabel.text = "Search Products"
+                searchProductsImageView.image = UIImage(named: "icn-tabbar-search.pdf")
+                searchProductsView.isHidden = false
+            } else {
+                autocompleteTableView.isHidden = false
+            }
+        }
+    }
+    var searchTerm = "" {
+        didSet {
+            searchBar.text = searchTerm
+            var _ = Timer.scheduledTimer(timeInterval: 0.5, target: self, selector: #selector(self.searchForProducts), userInfo: nil, repeats: false)
```

Comment:
We also want to save the timer and cancel the last one that was submitted so we don't just delay the search, but only do one search 0.5 seconds after the last keypress.